### PR TITLE
Add a tutorial for typesetting non-ASCII characters

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Address #2204.

**Preview**: https://pygmt-dev--3389.org.readthedocs.build/en/3389/tutorials/advanced/non_ascii_text.html

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: examples/tutorials/advanced/non_ascii_text.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# ruff: noqa: RUF001,RUF003
+"""
+Typesetting non-ASCII text
+--------------------------
+
+In addtion to ASCII printable characters, sometimes you may also want to typeset
```
Comment (yvonnefroehlich) [inline examples/tutorials/advanced/non_ascii_text.py:6]:
```suggestion
In addition to ASCII printable characters, sometimes you may also want to typeset
```

File: examples/tutorials/advanced/non_ascii_text.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# - **Similar-Looking Characters**: Be cautious when using characters that appear
+#   visually similar but are distinct. For example, ``Ω`` (OHM SIGN) and ``Ω`` (GREEK
+#   CAPITAL LETTER OMEGA) may look alike, but PyGMT only supports the latter. Using the
+#   incorrect character can lead to unexpected results. To avoid this, it's recommended
+#   to copy and paste characters from the :doc:`/techref/encodings` documentation.
+# - **Mix characters from different encodings**: As shown in the example above, you can
+#   mix characters from different encodings in the same text string. However, due to the
+#   limitations of the underlying PostScript language, you cannot mix characters from
+#   the "Adobe ISOLatin1+" and "ISO-8859-*x*" encodings in the same text string. For
+#   example, you cannot mix characters from the "Adobe ISOLatin1+" and "ISO-8859-2". If
```
Comment (yvonnefroehlich) [inline examples/tutorials/advanced/non_ascii_text.py:68]:
```suggestion
#   example, you cannot mix characters from "Adobe ISOLatin1+" and "ISO-8859-2". If
```

File: examples/tutorials/advanced/non_ascii_text.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#   you need to use characters from different encodings, you can use them in different
+#   PyGMT function/method calls.
+# - **Non-ASCII Characters in Text Files**: Non-ASCII characters are not supported if
```
Comment (yvonnefroehlich) [inline examples/tutorials/advanced/non_ascii_text.py:71]:
```suggestion
# - **Non-ASCII characters in text files**: Non-ASCII characters are not supported if
```

File: examples/tutorials/advanced/non_ascii_text.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    ],
+    font="18p,Helvetica",
+    justify="LM",
+)
+fig.show()
+
+# %%
+# Here are some important tips when using non-ASCII characters:
+#
+# - **Similar-Looking Characters**: Be cautious when using characters that appear
```
Comment (yvonnefroehlich) [inline examples/tutorials/advanced/non_ascii_text.py:59]:
```suggestion
# - **Similar-looking characters**: Be cautious when using characters that appear
```

File: examples/tutorials/advanced/non_ascii_text.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+#   limitations of the underlying PostScript language, you cannot mix characters from
+#   the "Adobe ISOLatin1+" and "ISO-8859-*x*" encodings in the same text string. For
+#   example, you cannot mix characters from the "Adobe ISOLatin1+" and "ISO-8859-2". If
+#   you need to use characters from different encodings, you can use them in different
+#   PyGMT function/method calls.
+# - **Non-ASCII Characters in Text Files**: Non-ASCII characters are not supported if
+#   you have them in a text file and pass it to ``Figure.text``. In this case, you may
+#   want to load the text file into :class:`pandas.DataFrame` and then pass it to the
+#   ``text`` parameter.
+#
```
Comment (yvonnefroehlich) [inline examples/tutorials/advanced/non_ascii_text.py:75]:
```suggestion
```

**Comment by seisman** [comment]
@yvonnefroehlich Is it OK to approve and merge?

File: examples/tutorials/advanced/non_ascii_text.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    text=[
+        "ABCD αβγδ ①②③ ➊➋➌",  # Mix characters from ISOLatin1+, Symbol and ZapfDingbats
+        "✈♥♦♣♠❛❜❝❞❨❩❪❫❬❭❮❯→↔",  # Non-ASCII characters from Adobe ZapfDingbats
+        "αβγδεζηθ⊗⊕∅⊃⊇⊄⊂⊆",  # Non-ASCII characters from Adobe Symbol
+        "±°ÀÁÂÃÄÅÈÌÒÙàèìòù",  # Non-ASCII characters from Adobe ISOLatin1+
+        "ABCDE12345!#$:;<=>?",  # ASCII only
+    ],
+    font="18p,Helvetica",
+    justify="LM",
+)
```
Comment (yvonnefroehlich) [inline examples/tutorials/advanced/non_ascii_text.py:53]: I'm wondering if it is better to have the same vertical order for the text strings in the code and the text lines  in the generated plot:

| inverse vertical order |  same vertical order |
| --- | --- |
| text_nonascii_inversorder | text_nonascii_sameorder |


```suggestion
fig.text(
    x=[0.2, 0.2, 0.2, 0.2, 0.2],
    y=[5, 4, 3, 2, 1],
    text=["ASCII:", "ISOLatin1+:", "Symbol:", "ZapfDingbats:", "Mixed:"],
    font="20p,Helvetica-Bold,red",
    justify="LM",
)
fig.text(
    x=[2, 2, 2, 2, 2],
    y=[5, 4, 3, 2, 1],
    text=[
        "ABCDE12345!#$:;<=>?",  # ASCII only
        "±°ÀÁÂÃÄÅÈÌÒÙàèìòù",  # Non-ASCII characters from Adobe ISOLatin1+
        "αβγδεζηθ⊗⊕∅⊃⊇⊄⊂⊆",  # Non-ASCII characters from Adobe Symbol
        "✈♥♦♣♠❛❜❝❞❨❩❪❫❬❭❮❯→↔",  # Non-ASCII characters from Adobe ZapfDingbats
        "ABCD αβγδ ①②③ ➊➋➌",  # Mix characters from ISOLatin1+, Symbol and ZapfDingbats
    ],
    font="18p,Helvetica",
    justify="LM",
)

```

**Review by yvonnefroehlich — APPROVED** [review]
Looks good!

**Pull request merged**